nixos/modules/programs/nix-required-mounts.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.nix-required-mounts;
+  package = pkgs.nix-required-mounts;
+  overridenPackage = package.override { inherit (cfg) allowedPatterns; };
+
+  Pattern = with lib.types;
+    submodule ({ config, name, ... }: {
+      options.onFeatures = lib.mkOption {
+        type = listOf str;
+        description =
+          "Which requiredSystemFeatures should trigger relaxation of the sandbox";
+        default = [ name ];
+      };
+      options.paths = lib.mkOption {
+        type = listOf path;
+        description =
+          "A list of glob patterns, indicating which paths to expose to the sandbox";
+      };
+    });
+
+  driverPaths = [
+    # symlinks in /run/opengl-driver/lib:
+    pkgs.addOpenGLRunpath.driverLink
+
+    # the actual userspace drivers in /nix/store:
+    config.hardware.opengl.package
+  ] ++ config.hardware.opengl.extraPackages;
+
+  defaults = {
+    opengl.onFeatures = package.allowedPatterns.opengl.onFeatures;
+    opengl.paths = package.allowedPatterns.opengl.paths ++ driverPaths;
+    cuda.onFeatures = package.allowedPatterns.cuda.onFeatures;
+    cuda.paths = package.allowedPatterns.cuda.paths ++ driverPaths;
```

Comment:
For instance, and it doesn't *have* to be done in this PR, but ROCm features/paths would go well here I believe.
[amdvlk/test.nix](https://github.com/NixOS/nixpkgs/blob/18e76671a16cebf6bcf52d9196d25a5a9f8c9ad7/pkgs/development/libraries/amdvlk/test.nix)
[clr/test-rocm-smi.nix](https://github.com/NixOS/nixpkgs/blob/18e76671a16cebf6bcf52d9196d25a5a9f8c9ad7/pkgs/development/rocm-modules/5/clr/test-rocm-smi.nix)

Reply:
CC @Flakebi as the maintainer of amdvlk test

What do you think?

Reply:
First time I saw this. Are there any instructions on how to run such a test? (I suppose I need more than just `nix-build -A nixosTests.<pkg>.tests`?)

If this works as advertised, it sounds like we should get rid of the impureTests. Thanks for cc’ing me!